### ISSUE TITLE
Query string is incorrectly cached

### ISSUE THREAD

**Issue by tpow** [MasoniteFramework/masonite#380]
I'm not sure why, but it appears that the query string is getting cached on GET requests, which means that tests against it give erroneous results. This is with Masonite 2.3.21.

The problem is visible in a view that shows request.all_query() and request.query_string(). In cases where the query_string is blank, all_query may have data from a previous request. This impacts use of request.query because it seems to rely on the dictionary from all_query.

**To Reproduce**
1. Create a test view and route. My view was something like this:
```python
    def test(self, request: Request, view: View, id: int):                                               
        my_var = request.query("var", "default")                             
        dd(request.all_query(), request.query_string(), my_var)
``` 
2. Hit the test view without a query string: '/test/'. All_query is an empty Dict and query_string is blank as expected.
3. Hit the test view with a query string '/test/?var=a'. All_query should give {'var': 'a'} (I think it actually gives {'var': ['a']} but I guess that allows for multiples, so not a big deal) and query_string is var=a.
4. Hit the test view with a changed query string '/test/?var=b'. All_query gives {'var': ['b']} and query_string is var=b.
5. Hit the test view with no query string again '/test/'. 

**Expected behavior**
All_query should be an empty dict like the first hit in step 2.

**Actual error**
All_query returns the dict of step 4. This happens despite the query_string being blank.

**Other notes**
If also displaying the request.query("var") you will get "b" in the final step. This is incorrect; the query string is empty.

I was using this to try to change the formatting in a view based on a query string variable. I'd prefer to be able to omit the query string to have the default, but once you hit it with the query string it unexpectedly caches it and gives you the previous request.

**Comment by josephmancuso** [comment]
I can't reproduce this. Is this happening in tests only or can you reproduce this in a browser?

**Comment by tpow** [comment]
I can reproduce it consistently in the browser. Just in case it might be browser specific, I tested it and was able to reproduce it in both Safari and Edge on a mac.

**Comment by josephmancuso** [comment]
thanks I'll try on those browsers. @girardinsamuel can you reproduce this?

**Comment by josephmancuso** [comment]
I actually just realized I was testing this on Masonite 3 which is fixed because of how the request class works in 3 .. Will test and fix on 2.3 before 3 gets released for you

**Comment by tpow** [comment]
Nice to hear that Masonite 3 has it fixed. I didn't know that you were making changes much beyond the ORM.
